test/index.ts

```diff
@@ -0,0 +1,66 @@
+import { expect } from "chai";
+import { BigNumber } from "ethers";
+import { ethers } from "hardhat";
+import { Token } from "../typechain/Token";
+
+describe("Token", () => {
+  let token: Token;
+  const decimals = 18;
+  const tokensQuantity = 100000;
+  const totalSupply = BigNumber.from(
+    String(tokensQuantity) + new Array(decimals).fill("0").join("")
```

Comment:
Myślę, że skoro już używasz `BigNumber` to można to załatwić liczbowo zamiast stringier :D `BigNumber.from(tokensQuantity).mul(BigNumber.from("10").pow(decimals))`